runtime/near-vm-logic/src/errors.rs

```diff
@@ -1,29 +1,52 @@
 use crate::dependencies::ExternalError;
 
-#[derive(Debug, Clone, PartialEq)]
+#[derive(Debug, Clone, PartialEq, Eq)]
 pub enum HostError {
     BadUTF16,
     BadUTF8,
     GasExceeded,
     GasLimitExceeded,
     BalanceExceeded,
     EmptyMethodName,
-    External(ExternalError),
     GuestPanic,
     IntegerOverflow,
-    InvalidIteratorIndex,
     InvalidPromiseIndex,
     CannotAppendActionToJointPromise,
     CannotReturnJointPromise,
     InvalidPromiseResultIndex,
     InvalidRegisterId,
     IteratorWasInvalidated,
     MemoryAccessViolation,
+    InvalidReceiptIndex,
```

Comment:
As @evgenykuzyakov suggested we should panic on `InvalidReceiptIndex`

Reply:
not panic but change StorageError to VMInternalFailure?